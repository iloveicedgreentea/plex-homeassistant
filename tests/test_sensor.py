"""Tests for the ezbeq sensors."""

from unittest.mock import AsyncMock, MagicMock, patch

import pytest
from syrupy import SnapshotAssertion

from homeassistant.const import STATE_UNKNOWN, Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from .conftest import setup_integration

from pytest_homeassistant_custom_component.common import MockConfigEntry, snapshot_platform

pytestmark = pytest.mark.asyncio


async def test_sensor_setup_and_update(
    hass: HomeAssistant,
    mock_ezbeq_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup of sensor and data update."""
    with patch("custom_components.ezbeq.PLATFORMS", [Platform.SENSOR]):
        await setup_integration(hass, mock_config_entry)

    for device_name in mock_ezbeq_client.device_info:
        entity_id = f"sensor.{device_name.name}_current_profile"
        entry = entity_registry.async_get(entity_id)
        assert entry

        state = hass.states.get(entity_id)
        assert state
        assert state.state == "Test Profile"

    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)

    # Simulate a data update
    mock_ezbeq_client.get_device_profile = MagicMock(return_value="New Test Profile")
    await hass.config_entries.async_reload(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    for device_name in mock_ezbeq_client.device_info:
        entity_id = f"sensor.{device_name.name}_current_profile"
        state = hass.states.get(entity_id)
        assert state
        assert state.state == "New Test Profile"

    # Simulate a data update with unavailable data
    mock_ezbeq_client.get_device_profile = MagicMock(return_value=None)
    await hass.config_entries.async_reload(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    state = hass.states.get(entity_id)
    assert state
    assert state.state == STATE_UNKNOWN
