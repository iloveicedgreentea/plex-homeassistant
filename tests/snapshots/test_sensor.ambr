# serializer version: 1
# name: test_sensor_setup_and_update[sensor.master2_current_profile-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.master2_current_profile',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Current profile',
    'platform': 'ezbeq',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'current_profile',
    'unique_id': '01J959G9VRJH1TFGKW53GSZ11N_master2_current_profile',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensor_setup_and_update[sensor.master2_current_profile-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'master2 Current profile',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.master2_current_profile',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'Test Profile',
  })
# ---
# name: test_sensor_setup_and_update[sensor.master_current_profile-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.master_current_profile',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Current profile',
    'platform': 'ezbeq',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'current_profile',
    'unique_id': '01J959G9VRJH1TFGKW53GSZ11N_master_current_profile',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensor_setup_and_update[sensor.master_current_profile-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'master Current profile',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.master_current_profile',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'Test Profile',
  })
# ---
